pasta/base/annotate_test.py

```diff
@@ -589,15 +589,26 @@ def test_indent_levels(self):
     self.assertEqual(1, fmt.get(foo, 'start_line'))
     self.assertEqual(0, fmt.get(foo, 'start_col'))
     self.assertEqual(2, fmt.get(foo, 'end_line'))
-    self.assertEqual(13, fmt.get(foo, 'end_col'))
+    self.assertEqual(12, fmt.get(foo, 'end_col'))
     self.assertEqual(2, fmt.get(bar_arg, 'start_line'))
     self.assertEqual(2, fmt.get(bar_arg, 'start_col'))
     self.assertEqual(2, fmt.get(bar_arg, 'end_line'))
-    self.assertEqual(12, fmt.get(bar_arg, 'end_col'))
+    self.assertEqual(11, fmt.get(bar_arg, 'end_col'))
     self.assertEqual(3, fmt.get(bar, 'start_line'))
     self.assertEqual(0, fmt.get(bar, 'start_col'))
     self.assertEqual(3, fmt.get(bar, 'end_line'))
-    self.assertEqual(9, fmt.get(bar, 'end_col'))
+    self.assertEqual(8, fmt.get(bar, 'end_col'))
+
+  def test_end_col(self):
+    src = 'a = b + c'
+    t = pasta.parse(src, astlib=astlib)
+    name_nodes = ast_utils.find_nodes_by_type(t, astlib.Name, astlib=astlib)
+    name_nodes.sort(key=lambda node: node.col_offset)
+    a, b, c = name_nodes
+
+    self.assertEqual(2, fmt.get(a, 'end_col'))
+    self.assertEqual(6, fmt.get(b, 'end_col'))
+    self.assertEqual(9, fmt.get(c, 'end_col'))
```

Comment:
This doesn't work in python2.7, which (tbh) isn't that big a deal, but pasta still supports it.

2.7:

```
"a + b"

NAME      'a' (1, 0), (1, 1)
OP        '+' (1, 2), (1, 3)
NAME      'b' (1, 4), (1, 5)
ENDMARKER ''  (2, 0), (2, 0)
```

3.9

```
"a + b"

NAME      'a' (1, 0), (1, 1)
OP        '+' (1, 2), (1, 3)
NAME      'b' (1, 4), (1, 5)
NEWLINE   ''  (1, 5), (1, 6)
ENDMARKER ''  (2, 0), (2, 0)
```

I'm not going to insist on reworking a bunch of token nonsense to get this to work, so proposing this instead:

```
# NB: 0 to support python2.7 which does not generate a NEWLINE token
self.assertIn(fmt.get(c, 'end_col'), (9, 0))
```

Reply:
Thanks for finding that and suggesting the fix!